### PR TITLE
571 update paired mr-us prostate demo

### PR DESCRIPTION
**Pull request by YipengHu**
# Description

Update the pretrained model for Paired prostate MR-ultrasound registration demo, together with the relevant config files and train/predict scripts.

Fixes #<issue_number>

## Type of change

What types of changes does your code introduce to DeepReg?

_Please check the boxes that apply after submitting the pull request._

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation Update (fix or improvement on the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (if none of the other choices apply)

## Checklist

_Please check the boxes that apply after submitting the pull request._

_If you're unsure about any of them, don't hesitate to ask. We're here to help! This is
simply a reminder of what we are going to look for before merging your code._

- [x] I have
      [installed pre-commit](https://deepreg.readthedocs.io/en/latest/contributing/setup.html)
      using `pre-commit install` and formatted all changed files. If you are not
      certain, run `pre-commit run --all-files`.
- [ ] My commits' message styles matches
      [our requested structure](https://deepreg.readthedocs.io/en/latest/contributing/commit.html),
      e.g. `Issue #<issue number>: detailed message`.
- [ ] I have updated the
      [change log file](https://github.com/DeepRegNet/DeepReg/blob/main/CHANGELOG.md)
      regarding my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: demos/paired_mrus_prostate/demo_data.py [hunk 1]
```diff
@@ -67,8 +67,8 @@
     shutil.rmtree(MODEL_PATH)
 os.mkdir(MODEL_PATH)
 
-ZIP_PATH = "paired_mrus_prostate-ckpt"
-ORIGIN = "https://github.com/DeepRegNet/deepreg-model-zoo/raw/master/paired_mrus_prostate-ckpt.zip"
+ZIP_PATH = "paired_mrus_prostate_1"
```
Comment (mathpluscode) [inline demos/paired_mrus_prostate/demo_data.py:70]: maybe a better name XD?

**Comment by mathpluscode** [comment]
Do you have any stats result showing the updated ckpt is better? predict script should give some metrics. @YipengHu

**Comment by YipengHu** [comment]
This is largely an update of pre-trained model. @s-sd you should be able to find a good example of visuals from this. #474 #565

File: demos/paired_mrus_prostate/demo_data.py (hunk 1 above)
Comment (YipengHu) [reply demos/paired_mrus_prostate/demo_data.py:70]: :D this is really for my internal record...

**Comment by mathpluscode** [comment]
another question is about how good do we want, for these models, as i guess the image size is still too small

**Comment by YipengHu** [comment]
> another question is about how good do we want, for these models, as i guess the image size is still too small

Good question - i'm digging out the old stats. This one is easier, we do have a good stats. For others, the improvement will be solely qualitative - but i think it would be obvious.

**Comment by mathpluscode** [comment]
> > another question is about how good do we want, for these models, as i guess the image size is still too small
> 
> Good question - i'm digging out the old stats. This one is easier, we do have a good stats. For others, the improvement will be solely qualitative - but i think it would be obvious.

yup, when dealing with the loading bug, we were comparing the stats and we considered 0.2 -> 0.6 in terms of dice means some improvement ^^

**Comment by YipengHu** [comment]
> > > another question is about how good do we want, for these models, as i guess the image size is still too small
> > 
> > 
> > Good question - i'm digging out the old stats. This one is easier, we do have a good stats. For others, the improvement will be solely qualitative - but i think it would be obvious.
> 
> yup, when dealing with the loading bug, we were comparing the stats and we considered 0.2 -> 0.6 in terms of dice means some improvement ^^

the numerical improvement was small for this case: dice on gland 067 -> 0.69 and probably not significant. I'd be reluctant to put more effort in this case though, considering the dummy data set.

**Comment by mathpluscode** [comment]
> > > > another question is about how good do we want, for these models, as i guess the image size is still too small
> > > 
> > > 
> > > Good question - i'm digging out the old stats. This one is easier, we do have a good stats. For others, the improvement will be solely qualitative - but i think it would be obvious.
> > 
> > 
> > yup, when dealing with the loading bug, we were comparing the stats and we considered 0.2 -> 0.6 in terms of dice means some improvement ^^
> 
> the numerical improvement was small for this case: dice on gland 067 -> 0.69 and probably not significant. I'd be reluctant to put more effort in this case though, considering the dummy data set.

which demo you mean, the prostate one? bcz you changed 3 files for paired prostate, plus 1 file for masked prostate.

**Review by mathpluscode — APPROVED** [review]

**Review by s-sd — APPROVED** [review]

**Comment by NMontanaBrown** [comment]
Hi @YipengHu @mathpluscode @s-sd, just catching up on new changes - appreciate the work that's going in to improve demos. However, changelog has not been updated, nor was the PR template used correctly - could we ensure as reviewers/PR submitters that we use it appropriately?

**Comment by mathpluscode** [comment]
> Hi @YipengHu @mathpluscode @s-sd, just catching up on new changes - appreciate the work that's going in to improve demos. However, changelog has not been updated, nor was the PR template used correctly - could we ensure as reviewers/PR submitters that we use it appropriately?

nice point.... maybe we can find a way to have the check automatically.

**Comment by YipengHu** [comment]
> > Hi @YipengHu @mathpluscode @s-sd, just catching up on new changes - appreciate the work that's going in to improve demos. However, changelog has not been updated, nor was the PR template used correctly - could we ensure as reviewers/PR submitters that we use it appropriately?
> 
> nice point.... maybe we can find a way to have the check automatically.

good point indeed - will add.
